### DISCUSSION TITLE
Way to integrate automated registry generation, based on LS data

### DISCUSSION THREAD

**Discussion by lifeart** [typed-ember/glint#205] [Ideas]
Usage example: https://github.com/lifeart/els-addon-glint/pull/3

Wondering, may it be useful for project migrations?

**Comment by chriskrycho** [comment]
Interesting idea, but given the way the resolver works, it doesn't actually need language server data to do it: assuming there is no custom resolver in play, the name can be generated automatically from the location on disk.

**Reply by lifeart** [reply]
this is exactly how LS detects it (based on disk location) and how sample is generated

**Reply by chriskrycho** [reply]
This may be a matter of framing/phrasing: "LS data" implies that you need a language server running to get the data, but in fact what you have is just a script which does something useful (which is triggered by the LS).

I'm curious what @dfreeman and @jamescdavis think; for my part I would not *at all* be inclined to pull this into Glint itself, but would be open to having it be extracted from ELS into a standalone tool that we can recommend.

**Reply by dfreeman** [reply]
I think I have the same question as Chris: if the input is files on disk, and the output is a file on disk, I'm not sure I understand where a language server enters into it? It certainly sounds like this could be useful as a standalone script/simple CLI for folks who are migrating!
